June 20 changelog (#1858)
sequinstream/sequin: CHANGELOG.md

This log is updated every Friday to track our feature releases and updates week by week. [Subscribe](https://sequinstream.com/#newsletter) to get updates in your inbox.

## June 20, 2025

<details>

<summary>Fixes and improvements</summary>

### Improved

> * The Postgres pool size is now configurable.

Reviewer: Which Postgres? Which commit is this referring to?

Author: https://github.com/sequinstream/sequin/commit/6827b3249c88c4dbe06895204eb2c984bd471af9

> * Improve Sequin's default performance.

Reviewer: ```suggestion
* Improvements to Sequin's performance.
* Improved how Sequin buffers messages when a replication slot is faster than a sink can handle, resulting in much smoother performance under heavy load.
```

* Increase the throughput of messages through filter functions.

> * Increase concurrency on SQS and SNS sinks as well as high availability deployments.

Reviewer: Drop

> * Improve experience when running Prometheus / Grafana locally.

Reviewer: Drop, i don't know what this is referring to?

Reviewer: Oh, referring to this? This is only relevant to Sequin devs

https://github.com/sequinstream/sequin/commit/d5ed7629f749fbe292242164f26f6444708127c3

> * Added default batch sizes for sinks to optimize performance.

Reviewer: ```suggestion
* Tweaked default batch sizes for sinks to optimize performance.
```

### Fixed

* Properly handle situations where a database or publication contain no tables.
* Handle errors when a deprecated YAML format is applied or planned.

> * Fix an issue where filter and routing functions weren't included in yaml exports.

Reviewer: ```suggestion
* Fix an issue where filter and routing functions weren't included in YAML exports.
```

This log is updated every Friday to track our feature releases and updates week by week. [Subscribe](https://sequinstream.com/#newsletter) to get updates in your inbox.

## June 20, 2025

> ### S2 Sinks
> 
> Sequin now supports [S2](https://s2.dev/blog/sequin) sinks!
> 

Reviewer: ```suggestion
### S2 Sink

Sequin now supports [S2](https://s2.dev/blog/sequin) as a sink destination!

```

<div align="center">
<img src="https://github.com/sequinstream/sequin/blob/main/docs/images/changelog/2025-06-20/s2-sink.png" alt="Sequin now natively integrates with S2" width="600" />
</div>

S2 is fully supported as a new sink in the console, API, and YAML. Read the [quickstart](https://sequinstream.com/docs/how-to/stream-postgres-to-s2#create-a-basin-and-stream) and [reference](https://sequinstream.com/docs/reference/sinks/s2) to learn how to get up and running with an S2 sink today.

#### When should you reach for S2

> S2 (a.k.a Stream Storage) is designs to be a cloud native, streaming primitive - like what S3 provides for blob storage, but for data in motion. In fact, S2 builds on this promise by using object storage for durability. This design provides throughput that rivals Kafka (125 MiBps / stream), at a lower cost, and with unlimited streams (no brokers and partitions). It comes with a simple API to append and read records from the stream.

Reviewer: ```suggestion
S2 (a.k.a. Stream Storage) is a streaming platform - like what S3 provides for blob storage, but for data in motion. Under the hood, S2 uses object storage for durability. This design provides throughput that rivals Kafka (125 MiBps / stream), at a lower cost, and with unlimited streams (no brokers and partitions). It comes with a simple API to append and read records from streams.
```
